### PR TITLE
Gnome Ring

### PR DESCRIPTION
**Pull request by adityashah1212**
A GPLv3 VoIP and video chat client. Website [https://ring.cx/](url)

### CODE REVIEW

File: cx.ring.Ring.json [hunk 1]
```diff
@@ -0,0 +1,142 @@
+{
+    "app-id": "cx.ring.Ring",
```
Comment (adityashah1212) [inline cx.ring.Ring.json]: Changed from cx.ring.RingGnome due to the fact that all interfaces and services are registered to  cx.ring.Ring.

File: cx.ring.Ring.json [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        "--talk-name=org.gnome.ScreenSaver",
+        "--talk-name=org.cinnamon.ScreenSaver",
+        "--talk-name=org.freedesktop.ScreenSaver",
+        "--talk-name=com.canonical.Unity.Session",
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        "--talk-name=com.canonical.indicator.application",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--own-name=cx.ring.RingGnome",
```
Comment (adityashah1212) [inline cx.ring.Ring.json]: This named is also needed by the application

**Comment by adityashah1212** [comment]
Could you guys please run a test build for this. I tested it x86_64, but I want to be sure of arm and aarch64

**Comment by nedrichards** [comment]
Test build: https://flathub.org/builds/#/builders/1/builds/633

**Comment by nedrichards** [comment]
Test build https://flathub.org/builds/#/buildrequests/3627?redirect_to_build=true (sorry for the previous one, I put in the wrong appid)

**Comment by nedrichards** [comment]
Looks like dbus-c++ needs a manual reconfigure to identify what an aarch64 machine is. Here's [an example of how to specify that](https://github.com/flathub/org.xiphos.Xiphos/blob/348705136cba52220ec7ee0d2e51355340173078/org.xiphos.Xiphos.json#L70) from another manifest.

**Comment by adityashah1212** [comment]
Please run a test build

**Comment by nedrichards** [comment]
Test build running here: https://flathub.org/builds/#/builders/1/builds/676

File: cx.ring.RingDaemon.json [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+            "sources":[
+                {
+                    "type": "git",
+                    "url": "https://github.com/open-source-parsers/jsoncpp",
+                    "branch": "1.8.3"
+                }
+            ]
+        },
+        {
+            "name": "dbus-c++",
```
Comment (nedrichards) [inline cx.ring.RingDaemon.json]: Sorry, I should've been more clear here. This also needs `"rm-configure": true,` in order to work on aarch64. I've just built down to that part on a scaleway aarch4 system so I know that'll work ;-)

Comment (adityashah1212) [reply cx.ring.RingDaemon.json]: Ok thanks will do it

**Comment by nedrichards** [comment]
Other than the rm-configure, the other failures aren't your fault @alexlarsson is investigating.

**Comment by adityashah1212** [comment]
Can you please trigger a build once libpulsecommon.so is fixed...

**Comment by nedrichards** [comment]
Rebuilding https://flathub.org/builds/#/builders/1/builds/703

**Comment by adityashah1212** [comment]
Arm build failed. Will fix after going home.

**Comment by nedrichards** [comment]
https://flathub.org/builds/#/builders/1/builds/710

**Comment by nedrichards** [comment]
OK, so 32 bit ARM works now, Aarch64 still fails though, I think you can safely ignore the x86-64 fail in the last build as that looks transient.

**Comment by adityashah1212** [comment]
aarch64 failure is because linker is not able to find libpulsecommon.so. This was something that was present in other ARCHs as well, but was fixed when Sdk was updated

**Comment by nedrichards** [comment]
bot, build cx.ring.Ring

File: cx.ring.libRingClient.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                "-accessibility",
+                "-dbus-linked",
+                "-fontconfig",
+                "-glib",
+                "-icu",
+                "-openssl-linked",
+                "-no-pch",
+                "-no-rpath",
+                "-no-separate-debug-info",
+                "-reduce-relocations",
```
Comment (nedrichards) [inline cx.ring.libRingClient.json]: This is different to the [KDE runtime's version](https://github.com/KDE/flatpak-kde-runtime/blob/master/org.kde.Sdk.json#L81) and seems to be causing a [build failure on aarch64](https://flathub.org/builds/#/builders/6/builds/880).

**Comment by nedrichards** [comment]
Ran another test build: https://flathub.org/builds/#/builders/1/builds/937

Fails on aarch64 with some stuff about reduce relocations in qt5. Your settings are [different to the KDE Runtime](https://github.com/KDE/flatpak-kde-runtime/blob/master/org.kde.Sdk.json#L81), which only enables that configure opt for x86-64, may be worth copying their configuration unless you need something that's different.

**Comment by adityashah1212** [comment]
Can you trigger another build. Sorry for late commit, was a little busy. Also I don't know if the build will go through, since I wasn't able to confirm it.

**Comment by nedrichards** [comment]
New build: https://flathub.org/builds/#/builders/1/builds/1025

**Comment by adityashah1212** [comment]
Please trigger another build, this should have fixed i386, x86_64 and arm builds. I have  made changes for aarch64 as well, but would like to see the status of the build

**Comment by nedrichards** [comment]
Building: https://flathub.org/builds/#/builders/1/builds/1036

**Comment by adityashah1212** [comment]
Please trigger a build, this should have fixed build issues with x86_64, i686 and arm
I have fixed the issue with aarch64, but that may not be the only one.

**Comment by Barthalion** [comment]
bot, build cx.ring.Ring

**Comment by Barthalion** [comment]
aarch64 still fails on qrencode

**Comment by adityashah1212** [comment]
Please trigger a build, I have moved qrencode to CMake. Hope this fixes issues with aarch64. If not I am disabling aarch64

File: cx.ring.Ring.json [hunk 5]
```diff
@@ -11,6 +11,7 @@
         "--share=network",
         "--device=all",
         "--share=ipc",
+        "--extension=org.freedesktop.Plaform.VAAPI=version=1.6",
```
Comment (TingPing) [inline cx.ring.Ring.json]: You don't need this, that extension point comes with the runtime you depend upon

**Comment by nedrichards** [comment]
bot, build cx.ring.Ring

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/cx.ring.Ring

**Pull request closed without merging**